source/Cosmos.System2/Graphics/Canvas.cs

```diff
@@ -317,32 +349,97 @@ public virtual void DrawRectangle(Pen pen, int x, int y, int width, int height)
             DrawLine(pen, xc, yc, xd, yd);
         }
 
-        public virtual void DrawFilledRectangle(Pen pen, int x_start, int y_start, int width, int height)
+        public virtual void DrawFilledRectangle(Pen pen, Point point, int width, int height = -1)
+        {
+
+            DrawFilledRectangle(pen, point.X, point.Y, width, height);
+
+        }
+
+        public virtual void DrawFilledRectangle(Pen pen, int x_start, int y_start, int width, int height = -1)
         {
+            if (height == -1)
+            {
+                height = width;
+            }
+
             for (int y = y_start; y < y_start + height; y++)
             {
                 DrawLine(pen, x_start, y, x_start + width - 1, y);
             }
         }
 
+        public virtual void DrawTriangle(Pen pen, Point point0, Point point1, Point point2)
+        {
+
+            DrawTriangle(pen, point0.X, point0.Y, point1.X, point1.Y, point2.X, point2.Y);
+
+        }
+
         public virtual void DrawTriangle(Pen pen, int v1x, int v1y, int v2x, int v2y, int v3x, int v3y)
         {
+
             DrawLine(pen, v1x, v1y, v2x, v2y);
             DrawLine(pen, v1x, v1y, v3x, v3y);
             DrawLine(pen, v2x, v2y, v3x, v3y);
+
+        }
+
+        public virtual void DrawArray(Color[] colors, Point point, int width, int height = -1)
+        {
+
+            DrawArray(colors, point.X, point.Y, width, height);
+
         }
-         
+        
+        public virtual void DrawArray(Color[] colors, int x, int y, int width, int height = -1)
```

Comment:
I imagine the idea of this is for draw Bitmaps / Fonts, it is a "naive" implementation as done for Clear() in Canvas, I think will make sense to do something similar to what I've done for Clear() in VBEScreen (this time instead of Fill / Memset() will be something as Memcpy() but the ASM code would similar).